server.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable @typescript-eslint/no-require-imports */
+const { createServer } = require("http");
+const { parse } = require("url");
+const next = require("next");
+const { Server } = require("socket.io");
+
+const dev = process.env.NODE_ENV !== "production";
+const app = next({ dev });
+const handle = app.getRequestHandler();
+
+app.prepare().then(() => {
+  const server = createServer((req, res) => {
+    const parsedUrl = parse(req.url, true);
+    handle(req, res, parsedUrl);
+  });
+
+  const io = new Server(server, { cors: { origin: "*" } });
+
+  io.on("connection", (socket) => {
+    console.log("A client connected info is:", socket.id);
+
+    socket.on("presentationEvent", (data) => {
+      console.log("socket presentationEvent:", data);
+      io.to(data.ref).emit("presentationEventMessage", data);
+    });
+
+    socket.on("error", (error) => {
+      console.log("A server error has occurred", error);
+    });
+
+    socket.on("disconnect", () => {
+      console.log("A client disconnected");
+    });
+  });
+
+  server.listen(3000, (err) => {
+    if (err) throw err;
+    console.log(`> Ready on ${process.env.PUBLIC_BASE_URL}`);
```

Comment:
If PUBLIC_BASE_URL is not defined it shows undefined. However, in other parts of the code it does use a default.

I think it would be nice to concentrate this management into a single file (e.g. constants.ts) so every module that needs to access them has a consistent value.

Like:
````ts
export const PORT = process.env.PORT ? Number(process.env.PORT) : 3000
export const PUBLIC_BASE_URL = process.env.PUBLIC_BASE_URL ?? `http://localhost:${PORT}`
```